dev/com.ibm.ws.persistence.mbean/src/com/ibm/ws/persistence/mbean/internal/DDLGenerationMBeanImpl.java

```diff
@@ -280,4 +285,37 @@ protected final String getDescription(MBeanOperationInfo opInfo, MBeanParameterI
         return description;
     }
 
+    @Trivial
+    private static final class DDLGenerationParticipantComparator implements Comparator<DDLGenerationParticipant> {
+
+        /** {@inheritDoc} */
+        @Override
+        public int compare(DDLGenerationParticipant o1, DDLGenerationParticipant o2) {
+            // Handle the case when two DDLGenerationParticipant instances are equal
+            if (o1.equals(o2)) {
+                return 0;
+            }
+
+            // If the DDLGenerationParticipant implementation types are the same
+            // and implement Comparable then defer to the compareTo method of the type.
+            boolean matchedTypes = o1.getClass() == o2.getClass();
+            if (matchedTypes && o1 instanceof Comparable && o2 instanceof Comparable) {
+                return ((Comparable) o1).compareTo(o2);
+            }
+
+            // If the DDLGenerationParticipant implementation types are different
+            // then order based on the type name (left / right branch of tree).  Therefore,
+            // all the SQL statements for each feature will always be grouped together.
+            if (!matchedTypes) {
+                return o1.getClass().getCanonicalName().compareTo(o2.getClass().getCanonicalName());
+            }
+
+            // Now we have to compare two DDLGenerationParticipant implementations of the same type,
+            // but do not provide a comparator. Always put the new participant after the current participant
+            // (o1 > o2) This will maintain insertion order which matches the previous behavior.
+            return 1;
```

Comment:
It seems like the right solution would be to make DDLGenerationParticipant extend Comparable, forcing all implementations to be comparable.
```suggestion
            // Now we have to compare two DDLGenerationParticipant implementations of the same type,
            // but do not provide a comparator. Always put the new participant after the current participant
            // (o1 > o2) This will maintain insertion order which matches the previous behavior.
            return 1; // TODO avoid this by making DDLGenerationParticipant extend Comparable and update implementations
```

Reply:
Yes I agree that it would be preferable for all DDLGenerationParticipant to be Comparable.  But I was trying to avoid affecting the current behavior of the other participants.  Adding a TODO here makes sense to consider changing this in the future.